### PR TITLE
add pre-commit hooks using golangci-lint

### PR DESCRIPTION
**Pull request by rdhariwal**
## Description

Replace pre-commit hooks with hooks from golangci-lint package

Hooks enabled:

```
     - gosec
     - golint
     - gofmt
     - goimports
     - govet
     - varcheck
     - typecheck
     - structcheck
     - deadcode
```
## Setup
1. Each hook can be tested individually by adding code that linter will catch. For example unused variables will be caught by `varcheck`. If you would like to test `gosec` consider adding a `rsa.GenerateKey(rand.Reader, 1024)` to a go file.

```sh
pre-commit run -a
```
## References

* [Pivotal story](https://www.pivotaltracker.com/story/show/165323373) for this change

### CODE REVIEW

File: pkg/tools/tools.go [hunk 1]
```diff
@@ -28,6 +28,8 @@ import (
 	_ "golang.org/x/tools/cmd/goimports"
 	// Install for pre-commit go-vet
 	_ "golang.org/x/tools/go/analysis/passes/shadow/cmd/shadow"
+	// Install for linting project files
+	_ "github.com/golangci/golangci-lint"
```
Comment (chrisgilmerproj) [inline pkg/tools/tools.go]: Make sure to remove the `gosec`, `go-imports`, `go-vet`, and `golint` tools from this file.

File: .circleci/config.yml [hunk 2]
```diff
@@ -16,30 +16,22 @@ executors:
     working_directory: ~/go/src/github.com/transcom/mymove
     docker:
       - image: trussworks/circleci-docker-primary:9ad70927655a601580c036d53456c70061c76cba
-        environment:
```
Comment (rdhariwal) [inline .circleci/config.yml]: @chrisgilmerproj should I revert this commit?

Comment (rdhariwal) [reply .circleci/config.yml]: I removed it through an interactive rebase :)

File: pkg/tools/tools.go (hunk 1 above)
Comment (chrisgilmerproj) [reply pkg/tools/tools.go]: Can you do me a favor and check if this is actually needed? My experiments with pre-commit make me think that it handles its own dependencies and we don't actually need to install it ourselves.

File: .circleci/config.yml [hunk 3]
```diff
@@ -257,6 +262,8 @@ jobs:
       - run: make get_gotools
       - run: make bin/callgraph
       - run: make server_generate
+      # this is so we can avoid go mod downloading and resulting in an error on a false positive
+      - run: go mod tidy || exit 0
```
Comment (chrisgilmerproj) [inline .circleci/config.yml]: should we add this to `scripts/pre-commit-go-mod`? And then we could call the script here.

File: .golangci.yml [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+     - deadcode
+  disable:
+    - unused #deprecated https://github.com/dominikh/go-tools/tree/master/cmd/unused
+    - errcheck #requires patching code
+    - gosimple # 20+ files need to be patched
+    - ineffassign # 20+ files need to be patched
+    - staticcheck # 30+files need to be patched
+issues:
+  exclude-rules:
+    # skip analyzing dutystationsloader until repeated tags from embedded types are fixed https://github.com/golang/go/issues/30846
```
Comment (chrisgilmerproj) [inline .golangci.yml]: Can you open a ticket in our icebox to address this once go1.13 lands?

Comment (rdhariwal) [reply .golangci.yml]: https://www.pivotaltracker.com/story/show/166386495

**Pull request merged**